src/core/auth.ts

```diff
@@ -67,12 +77,38 @@ export const auth = (client: ApolloClient<NormalizedCacheObject>): AuthSDK => {
    * @returns Promise resolved with AccountRegister type data
    */
   const register: AuthSDK["register"] = async opts =>
-    await client.mutate({
+    await client.mutate<RegisterMutation, RegisterMutationVariables>({
       mutation: REGISTER,
       variables: {
```

Comment:
Shouldn't we pass `channel` here from the config?

Reply:
To avoid forgetting that, perhaps we should always define it as required in the initial mutation and create a generic type that omits the channel from opts exposed to the consumer:
```
type OmitChannel<T> = Omit<T, "channel">;

const register: AuthSDK["register"] = (opts: OmitChannel<RegisterMutationVariables>) =>
```